basket/news/backends/ctms.py

```diff
@@ -103,11 +104,22 @@ def from_vendor(contact):
         elif group_name == "newsletters":
             # Import newsletter names
             # Unimported per-newsletter data: format, language, source, unsub_reason
-            newsletters = []
             for newsletter in group:
                 if newsletter["subscribed"]:
-                    newsletters.append(newsletter["name"])
-            data["newsletters"] = newsletters
+                    data.setdefault("newsletters", []).append(newsletter["name"])
+        elif group_name == "waitlists":
+            # Unimported per-waitlist data: source, extra fields...
+            for waitlist in group:
+                wl_name = waitlist["name"]
+                # Legacy waitlist format. For backward compatibility.
+                # See `to_vendor()`` and `waitlist_fields_for_slug()` for the inverse.
+                if wl_name == "vpn":
+                    wl_name = VPN_NEWSLETTER_SLUG
+                    data["fpn_country"] = waitlist["fields"]["geo"]
+                    data["fpn_platform"] = waitlist["fields"]["platform"]
+                if wl_name == "relay":
+                    data["relay_country"] = waitlist["fields"]["geo"]
+                data.setdefault("newsletters", []).append(wl_name)
```

Comment:
It's hard to tell from the diff here, but do you have tests for these `from_vendor` changes?

Reply:
There's only one test for `from_vendor`, and yes it covers the changes.
https://github.com/mozmeao/basket/blob/2a3b77d7d2fe919d8e54d8e90d84e4e4a4202584/basket/news/tests/test_ctms.py#L130-L133

But thanks for your comment, it made discover an issue about newsletters names, that I fixed in the latests commits